agent/dns_test.go

```diff
@@ -6069,6 +6072,88 @@ func TestDNS_AddressLookup(t *testing.T) {
 	}
 }
 
+func TestDNS_AddressLookupANY(t *testing.T) {
+	if testing.Short() {
+		t.Skip("too slow for testing.Short")
+	}
+
+	t.Parallel()
+	a := NewTestAgent(t, "")
+	defer a.Shutdown()
+	testrpc.WaitForLeader(t, a.RPC, "dc1")
+
+	// Look up the addresses
+	cases := map[string]string{
+		"7f000001.addr.dc1.consul.": "127.0.0.1",
+	}
+	for question, answer := range cases {
+		m := new(dns.Msg)
+		m.SetQuestion(question, dns.TypeANY)
+
+		c := new(dns.Client)
+		in, _, err := c.Exchange(m, a.DNSAddr())
+		if err != nil {
+			t.Fatalf("err: %v", err)
+		}
```

Comment:
Not blocking, but I have been converting any tests that use this style over to use `require.{NoError, Equal, ...}`. Not a big deal, but many of these `t.Fatalf` end up not providing enough information to debug. So as we write new tests we should probably prefer using `require` assertions instead.